### ISSUE TITLE
0.8.6 version breaks tests

### ISSUE THREAD

**Issue by steklopod** [danielroe/nuxt-vitest#238]
After upgrading to 0.8.6 ot 0.8.7 some of the tests fail with errors. And test task does not stop

```
TypeError: The "obj" argument must be an instance of Blob. Received an instance of Blob
    at new NodeError (node:internal/errors:405:5)
    at Function.createObjectURL (node:internal/url:1056:13)
```

```
[Vue warn]: Unhandled error during execution of setup function 
  at <Anonymous> 
  at <Anonymous ref="VTU_COMPONENT" > 
  at <VTUROOT>

Error: Test timed out in 2000ms.
If this is a long-running test, pass a timeout value as the last argument or configure it globally with "testTimeout".
```

```
(node:78788) PromiseRejectionHandledWarning: Promise rejection was handled asynchronously (rejection id: 1)
(Use `node --trace-warnings ...` to show where the warning was created)
(node:78788) PromiseRejectionHandledWarning: Promise rejection was handled asynchronously (rejection id: 1)
(Use `node --trace-warnings ...` to show where the warning was created)
[Vue warn]: Unhandled error during execution of setup function 
  at <Anonymous> 
  at <Anonymous ref="VTU_COMPONENT" > 
  at <VTUROOT>
(node:78788) PromiseRejectionHandledWarning: Promise rejection was handled asynchronously (rejection id: 1)
(Use `node --trace-warnings ...` to show where the warning was created)

Error: Test timed out in 2000ms.
```
<img width="318" alt="Снимок экрана 2023-07-01 в 12 13 42" src="https://github.com/danielroe/nuxt-vitest/assets/29500153/10a553ed-5486-479f-b57d-dcfdd69d117f">

___
Here is `vitest.config.ts`:
```ts
import { resolve as pathResolve } from 'node:path'
import { defineVitestConfig } from 'nuxt-vitest/config'
import AutoImport from 'unplugin-auto-import/vite'
import vuetify from 'vite-plugin-vuetify'

export default defineVitestConfig({
  plugins: [
    vuetify(),
    AutoImport({ imports: ['vue'], dirs: ['./composables/'] })
  ],
  resolve: { alias: { '@': pathResolve(__dirname), '~': pathResolve(__dirname) } },
  test: {
    dir: 'tests',
    testTimeout: 2000,
    environment: 'nuxt',
    globals: true,
    deps: { inline: ['vuetify'] },
    setupFiles: ['./tests/setup.ts'],
    coverage: {
      all: true,
      enabled: true,
      reporter: ['text', 'text-summary', 'cobertura'],
      exclude: ['api', '.nuxt', 'server', 'middleware', 'layouts', 'pages', 'tests', '*.config.ts', '*.d.ts', 'app.vue'],
      include: ['**/*.ts', '**/*.vue']
    }
  }
})
```

___

Here is an example of failed test:

```ts
import { describe, expect, test } from 'vitest'
import { mountSuspended } from 'vitest-environment-nuxt/utils'
import LogoMain from '~/components/main/LogoMain.vue'

describe('🪐 Main logo tests', () => {
  test('Main logo test', async () => {
    const component = await mountSuspended(LogoMain)

    const rootEl = component.find('a')
    expect(rootEl.attributes().href).toBe('/')

    const imgEl = component.find('img')
    const imgAttr = imgEl.attributes()
    expect(imgAttr.src).toContain('/static/img/logo/png/logo_text.png')
    expect(imgAttr.alt).toBeDefined()

    expect(component.html()).toMatchInlineSnapshot(`
     "<a href=\\"/\\"><img src=\\"https://colaba.online/static/img/logo/png/logo_text.png\\" height=\\"28\\" alt=\\"Go to main\\"></a>"
    `)
  })
})
```

Here is failed tests:

https://github.com/danielroe/nuxt-vitest/assets/29500153/2946f0b3-7daa-4873-ba01-0025e0529123

Here is working tests with version `0.8.5`

https://github.com/danielroe/nuxt-vitest/assets/29500153/6b824fcd-d1eb-4800-8168-6deb922016f8

**Comment by danielroe** [comment]
Would you provide a [reproduction](https://nuxt.com/docs/community/reporting-bugs/#create-a-minimal-reproduction)? 🙏

**Comment by steklopod** [comment]
Thank you Daniel for quick reaction!

Sure. Here it is: https://stackblitz.com/github/nuxt/starter/tree/v3-stackblitz?file=components%2FMapStudioMeta.vue

Just run `npm run test`


I figured out what causes the error: [Mapbox map component](https://vue-mapbox-gl.studiometa.dev/guide/introduction/)

But I don't know how to fix it.

<img width="858" alt="Снимок экрана 2023-07-02 в 12 52 02" src="https://github.com/danielroe/nuxt-vitest/assets/29500153/91a1255b-ca83-4115-bf10-17893d836c66">

**Comment by marr** [comment]
@steklopod That repro URL isn't correct. It's just pointing to the base starter template. I think you need to save the stackblitz which will fork it and allow us to see the files. Right now we can't see them.

**Comment by steklopod** [comment]
Oops. I'm sorry. Here it is: https://stackblitz.com/edit/github-e16yng?file=components%2FMapStudioMeta.vue

**Comment by marr** [comment]
@steklopod you're right that the mapbox component seems to break nuxt-vitest. It appears to be running client code on the server. I tried making the component a _*.client.vue_ component, but the issue persists.

I did notice some other issues with the code, and made some fixes here. You may be able to remove the plugins directory that I created but I'm not sure. 

[vitest-example.tar.gz](https://github.com/danielroe/nuxt-vitest/files/11929849/vitest-example.tar.gz)

**Comment by steklopod** [comment]
Thank you very much, Marr, for the attempt of help 🙏🏻, but it did not work if uncomment mapbox-component that you commented:

<img width="358" alt="Снимок экрана 2023-07-02 в 17 24 34" src="https://github.com/danielroe/nuxt-vitest/assets/29500153/be12edc6-6461-4a1b-a80d-4602b5e97458">

<img width="1282" alt="Снимок экрана 2023-07-02 в 17 28 41" src="https://github.com/danielroe/nuxt-vitest/assets/29500153/a681c962-0547-486a-9610-3014f4f93b9a">

**Comment by steklopod** [comment]
I am not sure but I think these changes in `0.8.6` version could brake 
<img width="1143" alt="Снимок экрана 2023-07-08 в 18 34 51" src="https://github.com/danielroe/nuxt-vitest/assets/29500153/655d02cd-0f6a-4a40-a7c2-c7c119c7cc76">

**Comment by danielroe** [comment]
Would you provide a runnable reproduction?

**Comment by steklopod** [comment]
Sure. I just downgraded `package.json` from `0.8.6` to `0.8.5`. 

[Here is the link with working tests](https://stackblitz.com/edit/github-e16yng-ysqzqu?file=tests%2Frendering%2Fmain-logo.test.ts)

<img width="799" alt="Снимок экрана 2023-07-08 в 20 42 11" src="https://github.com/danielroe/nuxt-vitest/assets/29500153/4524db3f-90bb-450d-b239-21002ccb23a5">

**Comment by steklopod** [comment]
The same is for version `0.10.1`

Maybe there is a mechanism of ignoring/excluding a tag from rendering in tests?

**Comment by danielroe** [comment]
Or is your point that the blog setup bit is required?

**Comment by steklopod** [comment]
Thank you Daniel for the help 🙏🏻. I found how to fix this.

> The link you provided has not working test:
<img width="598" alt="Снимок экрана 2023-07-20 в 20 11 26" src="https://github.com/danielroe/nuxt-vitest/assets/29500153/717a67c5-41da-4a2e-a381-8e29a358d334">

> Step 0

I updated dependencies and simplified `vitest.config.ts` like in your link and test did not pass:
<img width="607" alt="Снимок экрана 2023-07-20 в 20 13 39" src="https://github.com/danielroe/nuxt-vitest/assets/29500153/ee4f0e8c-af30-4f8b-8401-bb737ab979ff">

> Step 1

I installed `jsdom`

> Step 2

Added 1 row in `vitest.config.ts`:

```ts
    environmentOptions: { nuxt: { domEnvironment: 'jsdom' } },
```

This is not covered by doc. I found it inside some other issue of this repo.

<img width="517" alt="Снимок экрана 2023-07-20 в 20 31 15" src="https://github.com/danielroe/nuxt-vitest/assets/29500153/6fbe5f8f-f31b-4e66-8bf9-c52b463cae3d">

Tests passed:
<img width="539" alt="Снимок экрана 2023-07-20 в 20 32 16" src="https://github.com/danielroe/nuxt-vitest/assets/29500153/adb752b3-1596-4223-9b9b-326e92a7c6ca">

You can check yourself:
https://stackblitz.com/edit/github-e16yng?file=vitest.config.ts

**Comment by danielroe** [comment]
Oh yes, there was a recent breaking change where you need to install either happy-dom or jsdom. If you install happy-dom you do not need to change the config.

**Comment by steklopod** [comment]
With `happy-dom` it does not work. The error is attached in first message of the issue.
Only this row makes tests working:
```js
environmentOptions: { nuxt: { domEnvironment: 'jsdom' } }
```
